geode-assembly/src/integrationTest/java/org/apache/geode/tools/pulse/PulseSecurityConfigOauthProfileTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.tools.pulse;
+
+import static org.apache.geode.test.junit.rules.HttpResponseAssert.assertResponse;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Properties;
+
+import org.apache.http.HttpResponse;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.examples.SimpleSecurityManager;
+import org.apache.geode.test.junit.categories.PulseTest;
+import org.apache.geode.test.junit.rules.GeodeHttpClientRule;
+import org.apache.geode.test.junit.rules.LocatorStarterRule;
+
+@Category({PulseTest.class})
+public class PulseSecurityConfigOauthProfileTest {
+  // this test just makes sure the property file in the locator's working dir
+  // gets properly read and used in the oauth security configuration
+
+  @ClassRule
+  public static LocatorStarterRule locator =
+      new LocatorStarterRule().withHttpService()
+          .withSecurityManager(SimpleSecurityManager.class)
+          .withProperty("security-auth-token-enabled-components", "pulse");
+
+  private static File pulsePropertyFile;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    // copy the pulse.properties to the locator's working dir. Pulse will use the locator's working
+    // dir as classpath to search for this property file
+    pulsePropertyFile = new File(locator.getWorkingDir(), "pulse.properties");
+    Properties properties = new Properties();
+    properties.setProperty("pulse.oauth.provider", "uaa");
+    properties.setProperty("pulse.oauth.clientId", "pulse");
+    properties.setProperty("pulse.oauth.clientSecret", "secret");
+    // have the authorization uri point to a known uri that locator itself can serve
+    properties.setProperty("pulse.oauth.authorizationUri",
+        "http://localhost:" + locator.getHttpPort() + "/management");
```

Comment:
It looks as if this uses the /management API itself to somehow mimic UAA (or some other OAuth server). How does that work?

Reply:
this changeset only makes sure pulse can be configured to talk to an authorizationUri. Whatever happens after that, is going to be in Joris's PR.

Reply:
As the test comment states "this test just makes sure the property file in the locator's working dir
gets properly read and used in the oauth security configuration". so it only makes sure the configuration in the property file gets used by Pulse, exactly where that URI points to, is irrelavant.